TicketsDemo.Mongo/Repositories/MongoTrainRepository.cs

```diff
@@ -0,0 +1,53 @@
+﻿using MongoDB.Driver;
+using System.Collections.Generic;
+using System.Linq;
+using MongoDB.Bson;
+using TicketsDemo.Data.Entities;
+using TicketsDemo.Data.Repositories;
+
+namespace TicketsDemo.Mongo
+{
+    public class MongoTrainRepository : ITrainRepository
+    {
+
+        public List<Train> GetAllTrains()
+        {
+            var ctx = new TicketsContext();
+            var train = ctx.Trains.Find(new BsonDocument()).ToList();
```

Comment:
here Carriage is not identified for place objects and
and Train is not identified for Cariage objects, so do this the same inverse assotiations assignments done in GetTrainDetails